core/src/main/java/cn/edu/tsinghua/iot/benchmark/client/generate/GenerateDataMixClient.java

```diff
@@ -140,9 +140,10 @@ private boolean ingestionOperation() {
           }
         }
         for (int j = 0; j < innerLoop; j++) {
-          Batch batch = dataWorkLoad.getOneBatch();
+          IBatch batch = dataWorkLoad.getOneBatch();
           if (checkBatch(batch)) {
             dbWrapper.insertOneBatch(batch);
+            batch.finishCheck();
```

Comment:
From my perspective, the finish check of the batch should be part of `insertOneBatch` method. I think the agreement of `insertOneBatch` method should be we insert all of data in batch into database

Reply:
Indeed. 
However, there are too many implementations of `insertOneBatch`, so I add `insertOneBatchWithCheck` method to `IDatabase` interface